firefly-api/src/providers.rs

```diff
@@ -30,25 +33,25 @@ impl FireflyProvider {
         }
     }
 
-    pub async fn client(&self, wallet_key: &str) -> anyhow::Result<Client> {
-        Client::new(
+    pub async fn client(&self, wallet_key: &str) -> Result<Client, anyhow::Error> {
+        let client = Client::new(
             wallet_key,
-            self.deploy_service_url.clone(),
-            self.propose_service_url.clone(),
+            &self.deploy_service_url,
+            &self.propose_service_url,
         )
-        .await
-        .context("Failed to create Firefly client")
+        .await;
+        client.context("Failed to create Firefly client: {err}")
```

Comment:
context doesnt work as format! macro. in this case you will have just `: {err}` appended to the end without any substitution
```suggestion
        client.context("Failed to create Firefly client")
```

Reply:
done